full_bubble_model.py

```diff
@@ -282,75 +407,154 @@ def f(t, x, alfa_M, T_inf, P_v, P_inf, mu_L, surfactant):
+        integrand_th = -(p * (1 + R_dot / c_L) + R / c_L * p_dot) * V_dot
+        integrand_v = 16.0 * np.pi * mu_L * (R * R_dot*R_dot + R * R * R_dot * dxdt[1] / c_L)
+        integrand_r = 4.0 * np.pi / c_L * R * R * R_dot * (R_dot * p + p_dot * R - 0.5 * par.rho_L * R_dot * R_dot * R_dot - par.rho_L * R * R_dot * dxdt[1])
+
+        dxdt[-1]=(integrand_th + integrand_v + integrand_r)
+    else:
+        dxdt[-1]=0.0
+    
     return dxdt
 
 
 """________________________________Solving________________________________"""
 
-# This funfction solves the differential equation, and returns the numerical solution
-# Error codes:
-    # 0: succecfully solved with LSODA solver
-    # 1: LSODA solver didn't converge, but Radau solver worked
-    # 2: LSODA solver timed out, but Radau solver worked
-    # 3: LSODA solver had a fatal error, but Radau solver worked
-    # 4: LSODA solver failed, Radau solver didn't converge (NO SOLUTION!)
-    # 5: LSODA solver failed, Radau solver timed out (NO SOLUTION!)
-    # 6: LSODA solver failed, Radau solver had a fatal error (NO SOLUTION!) 
 def solve(cpar, t_int=np.array([0.0, 1.0]), LSODA_timeout=30, Radau_timeout=300):
+    """
+    This funfction solves the differential equation, and returns the numerical solution.
+    Parameters:
+     * cpar: control parameters
+     * t_int: time interval
+     * LSODA_timeout: timeout for LSODA solver
+     * Radau_timeout: timeout for Radau solver
+
+    Returns:
+     * num_sol: numerical solution. use num_sol.t and num_sol.y to get the time and the solution
+     * error_code: see de.error_codes: dict, de.get_errors()
+     * elapsed_time: elapsed time
+    """
     error_code = 0
     start = time.time()
     if not 'P_v' in cpar:
         cpar.P_v = VapourPressure(T=cpar.T_inf) # [Pa]
     if not 'mu_L' in cpar:
         cpar.mu_L = Viscosity(T=cpar.T_inf) # [Pa]
-    IC = InitialCondition(cpar)
+    ex_args = []
+    for name, unit in zip(excitation_args, excitation_units):
+        if name in cpar:
+            ex_args.append(cpar[name])
+        else:
+            ex_args.append(0.0)
+            print(colored(f'Warning! Pressure excitation argument \'{name} [{unit}]\' is not in cpar. 0 is used instead. ', 'yellow'))
+    ex_args = np.array(ex_args, dtype=np.float64)
+    IC, lowpressure_error, lowpressure_warning = InitialCondition(cpar, enable_evaporation)
+    if lowpressure_error:
+        error_code += 100
+        return None, error_code, 0.0
+    elif lowpressure_warning:
+        error_code += 200
     
-    # solving d/dt x=f(t, x, alfa_M, T_inf, P_v, P_inf, mu_L, surfactant)
-    try:
-        num_sol = func_timeout(LSODA_timeout, solve_ivp, kwargs=dict(fun=f, t_span=t_int, y0=IC, method='LSODA', atol = 1e-10, rtol=1e-10, args=(cpar.alfa_M, cpar.T_inf, cpar.P_v, cpar.P_inf, cpar.mu_L, cpar.surfactant)))
+    # solving d/dt x=f(t, x, cpar)
+    try: # try-catch block
+        num_sol = func_timeout( # timeout block
```

Comment:
this line is much easier to read. Also error codes are now reimagined to hold more information. Different digits represent errors/warnings for LSODE, Radau, and in general. See below